src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCISharedBuildJobQueueService.java

```diff
@@ -51,9 +51,16 @@ public class LocalCISharedBuildJobQueueService {
 
     private final ProgrammingExerciseRepository programmingExerciseRepository;
 
+    // map of build jobs currently being processed accross all nodes
     private final IMap<Long, LocalCIBuildJobQueueItem> processingJobs;
 
-    private final FencedLock lock;
+    private int localProcessingJobs = 0;
```

Comment:
Would it make sense to use an `AtomicInteger` instead to not accidentally access it concurrently?